src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java

```diff
@@ -296,7 +298,7 @@ public FeatureCollectionResponse run(GetFeatureRequest request)
 
         boolean isNumberMatchedSkipped = false;
         int count = 0; // should probably be long
-        int totalCount = 0;
+        BigInteger totalCount = BigInteger.ZERO;
```

Comment:
That's what the EMF interface takes, complain with the EMF code generator, not me :)